fe/src/main/java/org/apache/doris/planner/OlapScanNode.java

```diff
@@ -130,6 +138,90 @@ public void setForceOpenPreAgg(boolean forceOpenPreAgg) {
         this.forceOpenPreAgg = forceOpenPreAgg;
     }
 
+    public Collection<Long> getSelectedPartitionIds() {
+        return selectedPartitionIds;
+    }
+
+    /**
+     * This method is mainly used to update scan range info in OlapScanNode by the new materialized selector.
+     * Situation1:
+     * If the new scan range is same as the old scan range which determined by the old materialized selector,
+     * the scan range will not be changed.
+     * <p>
+     * Situation2: Scan range is difference. The type of table is duplicated.
+     * The new scan range is used directly.
+     * The reason is that the old selector does not support SPJ<->SPJG, so the result of old one must be incorrect.
+     * <p>
+     * Situation3: Scan range is difference. The type of table is aggregated.
+     * The new scan range is different from the old one.
+     * If the test_materialized_view is set to true, an error will be reported.
+     * The query will be cancelled.
+     * <p>
+     * Situation4: Scan range is difference. The type of table is aggregated. `test_materialized_view` is set to false.
+     * If the enable_new_mv_selector is set to false, the result of the old version selector will be selected.
```

Comment:
In situation4, i think if `enable_new_mv_selector` is set to false, this `updateScanRangeInfo()` method won' t be called at all. And even the MaterializedViewSelector won't be called at all.